### ISSUE TITLE
Configure Azure KeyVault backed secret scope

### ISSUE THREAD

**Issue by Simon-Bru** [databricks/databricks-cli#262]
Hi,
I'm working on Databricks for a big data/machine learning project. I already configured an azure keyvault backed secret scope from the user interface. However, I am trying now to setup some CI/CD on the project and it includes an automatic configuration of Databricks and the secret scope.

The create-scope command is very limited, it would be great if we could specify the resource ID and DNS name of the Azure Keyvault for Azure hosted databricks workspaces.

I've noticed a lot of limitation in term of workspace configuration automation, I hope you can consider adding this feature in the next release. 

Please let me know if you are interested about improving those limitations (like token generation for instance).

Thanks for all your work 
Simon

**Comment by m1nkeh** [comment]
really big issue, watching with eagerness!

**Comment by nkangdatabricks** [comment]
Hi, thanks for reporting! Supporting AKV secret scope in cli is blocked by AAD authentication in the databricks cli which is on our roadmap.

**Comment by Simon-Bru** [comment]
Looking forward to this ! It would indeed solve all my problems.

**Comment by Simon-Bru** [comment]
Hello,
Yes AAD Authentication is now in preview, but only works with the databricks REST API. I had the documentation during a Databricks workshop at MS Paris. But the Rest API doesn't support key-vault backed secret scopes creation either... As the rest API now supports that authentication scheme in preview, I guess it can be implemented for the CLI and help that issue getting resolved. 
Looking forward for updates about this feature.
++

**Comment by rafaelpierre** [comment]
Any updates on this? :)

**Comment by annashres** [comment]
Hi - AAD auth via the API will be available publically in this coming month. We can then add support for the CLI. 
This will enable users to access AKV secrets via the CLI, although creating an AKV secret scope via the CLI does need some additional work. We are targeting that in Q3 2020.

**Comment by stack247** [comment]
We have encountered the same need, generally, to automate creation and deployment of workspace, clusters and jobs.
One of the issues is what everyone says here: creating key vault backed scope.
Some others that I could think of is
- Create access token programatically or use username / password to access databricks (`databricks configure`).
- Option to create a new job but don't create cluster if it already exists.

Does anyone know if any of these is in the roadmap and when it will be available? Thanks.

**Comment by rom1gal** [comment]
Hi ! 
Any updates ? :-)

**Comment by bigdatamoore** [comment]
I am interested in updates to this as well.

**Comment by phanvinh0526** [comment]
I have same demand, waiting for any update on this outstanding issue!

**Comment by sagarkul** [comment]
as per the documentation, 
https://docs.microsoft.com/en-us/azure/databricks/security/secrets/secret-scopes#--create-an-azure-key-vault-backed-secret-scope
we can do this using 

- Create the Azure Key Vault scope:


`databricks secrets create-scope --scope <scope-name> --scope-backend-type AZURE_KEYVAULT --resource-id <azure-keyvault-resource-id> --dns-name <azure-keyvault-dns-name>`

y default, scopes are created with MANAGE permission for the user who created the scope. If your account does not have the Azure Databricks Premium Plan, you must override that default and explicitly grant the MANAGE permission to the users (all users) group when you create the scope:

`databricks secrets create-scope --scope <scope-name> --scope-backend-type AZURE_KEYVAULT --resource-id <azure-keyvault-resource-id> --dns-name <azure-keyvault-dns-name> --initial-manage-principal users`

To create a scope using the Databricks CLI:

`databricks secrets create-scope --scope <scope-name>`


explicitly grant the MANAGE permission to “users” (all users) when you create the scope:

`databricks secrets create-scope --scope <scope-name> --initial-manage-principal users
`

**Comment by bigdatamoore** [comment]
@sagarkul thanks, yes that is correct.   This works now.  

There is one limitation still, which is that in Azure this cannot be run from a service principal, which makes DevOps automation tricky.

**Comment by m1nkeh** [comment]
> @sagarkul thanks, yes that is correct. This works now.
> 
> There is one limitation still, which is that in Azure this cannot be run from a service principal, which makes DevOps automation tricky.

are you sure? `databricks configure --aad-token` worked for me last week when I wrote a wee little DevOps pipeline to deploy some bits...

**Comment by smdibm** [comment]
@m1nkeh , Good to hear that, I am also facing the same issue in Azure DevOps with terraform, can you share more details to create secret scope backed by Keyvault?

**Comment by vivsri** [comment]
any luck with automation around this as it’s still complains about invalid token

**Comment by vivsri** [comment]
> 

how did you generate & pop up the token for the pipeline

**Comment by nfx** [comment]
Duplicate of #338
